### PR TITLE
Enable IMDSv2 on all instances

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Enabling IMDSv2 on all nodes by default in order to have more secure calls. Read more about it https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/infra/infra-stack.ts [hunk 1]
```diff
@@ -5,35 +5,35 @@ The OpenSearch Contributors require contributions made to
 this file be licensed under the Apache-2.0 license or a
 compatible open source license. */
 
+import {
```
Comment (peterzhuamazon) [inline lib/infra/infra-stack.ts:8]: This is just reorder of all the imports right?

File: test/os-cluster.test.ts [hunk 2]
```diff
@@ -71,6 +71,11 @@ test('Test Resources with security disabled multi-node default instance types',
       Ref: 'managerNodeAsgInstanceProfile1415C2CF',
     },
   });
+  infraTemplate.hasResourceProperties('AWS::AutoScaling::LaunchConfiguration', {
+    MetadataOptions: {
+      HttpTokens: 'required',
```
Comment (peterzhuamazon) [inline test/os-cluster.test.ts:76]: Is httptokens the same as imdsv2?

**Review by peterzhuamazon — APPROVED** [review]
Will let @rishabh6788 do another round of review.
Add some comments.

Thanks.

File: lib/infra/infra-stack.ts (hunk 1 above)
Comment (gaiksaya) [reply lib/infra/infra-stack.ts:8]: Yes my IDE does that!

File: test/os-cluster.test.ts (hunk 2 above)
Comment (gaiksaya) [reply test/os-cluster.test.ts:76]: Yes! When you enable IMDSv2, the metadata retrieval needs to be authenticated.

**Pull request merged**